compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1263,6 +1268,48 @@ fn project<'cx, 'tcx>(
     }
 }
 
+/// The first thing we have to do is scan through the parameter
+/// environment to see whether there are any projection predicates
+/// there that can answer this question.
```

Comment:
copy 🍝 

maybe include some code example explaining what this funciton finds in the documentation, too